public/pages/Dashboard/Components/AnomaliesLiveChart.tsx

```diff
@@ -0,0 +1,293 @@
+          {
+            range: {
+              [AD_DOC_FIELDS.ANOMALY_GRADE]: {
+                gt: 0.0,
+              },
+            },
+          },
+        ],
+        must_not: [
+          {
+            exists: {
+              field: AD_DOC_FIELDS.ERROR,
+            },
+          },
+        ],
+      },
+    },
+    sort: {
+      [AD_DOC_FIELDS.DATA_START_TIME]: SORT_DIRECTION.DESC,
+    },
+    collapse: {
+      field: AD_DOC_FIELDS.DETECTOR_ID,
+    },
+  };
+
+  const getRecentAnomalyResultQuery = {
+    range: {
+      [AD_DOC_FIELDS.DATA_START_TIME]: {
+        gte: 'now-30m',
+      },
+    },
+    size: 30,
+    sortField: AD_DOC_FIELDS.DATA_START_TIME,
+    from: 0,
+    sortDirection: SORT_DIRECTION.DESC,
+  };
+
+  const timeNowLineStyle = {
+    line: {
+      strokeWidth: 1,
+      stroke: '#3F3F3F',
+      dash: [1, 2],
+      opacity: 0.8,
+    },
+  };
+
+  const [liveTimeRange, setLiveTimeRange] = useState<LiveTimeRangeState>({
+    startDateTime: moment().subtract(30, 'minutes'),
+    endDateTime: moment(),
+  });
+
+  const [timer, setTimer] = useState();
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [liveAnomalyData, setLiveAnomalyData] = useState([] as object[]);
+
+  const getLiveDetectors = async () => {
+    if (!props.allDetectorsSelected) {
+      return props.selectedDetectors;
+    }
+    const detectorArray = [] as Detector[];
+    const searchResponse = await dispatch(
+      searchES(getRecentAnomalousDetectorsQuery)
+    );
+    const searchResults = searchResponse.data.response;
+    const numDetectors = get(searchResults, 'hits.total.value', 0);
+
+    if (numDetectors < 1) {
+      return detectorArray;
+    }
+    const liveDetectorIds: string[] = get(searchResults, 'hits.hits', []).map(
+      (result: any) => result._source.detector_id
+    );
+
+    for (let detectorId of liveDetectorIds) {
+      const getDetectorResp = await dispatch(getDetector(detectorId));
+      detectorArray.push(getDetectorResp.data.response);
+    }
+
+    return detectorArray;
+  };
+
+  const getLiveAnomalyData = async (currentDetectors: Detector[]) => {
+    // clear previous result
+    const currentLiveAnomalyData = [] as object[];
+    for (let detector of currentDetectors) {
+      const resp = await dispatch(
+        getDetectorResults(detector.id, getRecentAnomalyResultQuery)
+      );
+      currentLiveAnomalyData.push({
+        detector: detector,
+        results: resp.data.response.results,
+      });
+    }
+
+    setLiveAnomalyData(currentLiveAnomalyData);
+
+    setLiveTimeRange({
+      startDateTime: moment().subtract(30, 'minutes'),
+      endDateTime: moment(),
+    });
+    setIsLoading(false);
+  };
+
+  useEffect(() => {
+    async function initializeAnomaliesChart() {
+      const liveDetectors = await getLiveDetectors();
+      getLiveAnomalyData(liveDetectors);
+      setTimer(
+        setInterval(getLiveAnomalyData, MIN_IN_MILLI_SECS, liveDetectors)
+      );
```

Comment:
Let's add TODO to make this smart. We can stop this and save backend calls if the browser tab is not focused. i.e 
let's say I am on this page but when switch tab , all the calls we make will be waste and not really useful, can we pause it and start it again when user comes back  ? Another way we can introduce back-off mechanism.

Reply:
hmmm, that is good point. For now we don;t have requirement to support multi tabs, but this definitely should be considered in the future

Reply:
I will add TODO for it.